cpp/src/binaryop/jit/code/operation.cpp

```diff
@@ -26,10 +26,11 @@ const char* operation =
 R"***(
 #pragma once
     #include "traits.h"
+    #include <cudf/types.h>
 
     struct Add {
         template <typename TypeOut, typename TypeLhs, typename TypeRhs>
-        static TypeOut operate(TypeLhs x, TypeRhs y) {
+        static TypeOut operate(TypeLhs x, gdf_bool8 x_valid, TypeRhs y, gdf_bool8 y_valid) {
```

Comment:
So these parameters x_valid and y_valid are unused in most operators. I don't like adding them everywhere they aren't used.

Reply:
Agree 100%, but tbh I was struggling getting cpp to build at all, so I just consider this a checkpoint commit. I intend to separate the comparison operators since they are the only ones (that I know of atm) which need the validity bit.

Reply:
@harrism is `gdf_bool8` an appropriate type to use here, or is there something better suited?

Reply:
`bool`?